### PR TITLE
Implementation of Conditional Gradient Optimizer. Add files via upload

### PR DESCRIPTION
**Pull request by pkan2**
Hello,

I have uploaded the file "conditional_gradient.py" and "conditional_gradient_test.py" file into this directory. They are the implementation of conditional gradient optimizer and the test file of it. The contributors are Pengyu Kan and Vishnu sai rao suresh Lokhande, whose github name is lokhande-vishnu.

Thank you!

Sincerely,

Pengyu Kan

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Hi @pkan2! Thanks for the PR.

Just FYI typically we [request that you create an issue before submitting a PR](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#pull-requests) so that we can discuss beforehand.

For other's visibility this optimizer is described in this paper (I have not yet had a chance to read it):
https://arxiv.org/pdf/1803.06453.pdf

A quick glance at the PR though shows a number of issues. Please review the [TFA Optimizer requirements](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#standard-api)

Some examples:
* All TFA optimizers must subclass a Keras base optimizer
* We only allow public TF api usage (i.e. [tf.dtypes.cast](https://www.tensorflow.org/api_docs/python/tf/dtypes/cast) instead of `math_ops.cast`)

It may help to look at some of the other optimizers already in the repo.

**Comment by lokhande-vishnu** [comment]
Hi @seanpmorgan! I'm one of the authors of the paper and a contributor to the code. Your comments were very helpful. We'll modify our code accordingly and open a new issue!
Thanks

**Comment by facaiy** [comment]
Thank you, Vishnu, Look forward to your new issue :-) Let's close the PR, and of course, feel free to reopen it if necessary.
